pycobertura/cobertura.py

```diff
@@ -376,27 +376,47 @@ def diff_missed_lines(self, filename):
 
     def files(self, ignore_regex=None):
         """
-        Return `self.cobertura2.files()`.
+        Return the total of all files we're comparing.
         """
-        return self.cobertura2.files(ignore_regex)
+        f = list(
+            set(
+                self.cobertura2.files(ignore_regex)
+                + self.cobertura1.files(ignore_regex)
+            )
+        )
+        f.sort()
+        return f
 
     def file_source(self, filename):
         """
         Return a list of namedtuple `Line` for each line of code found in the
         given file `filename`.
 
         """
+        nonexistent = True
         if self.cobertura1.has_file(filename) and self.cobertura1.filesystem.has_file(
             filename
         ):
             lines1 = self.cobertura1.source_lines(filename)
             line_statuses1 = dict(self.cobertura1.line_statuses(filename))
+            nonexistent = False
         else:
             lines1 = []
             line_statuses1 = {}
 
-        lines2 = self.cobertura2.source_lines(filename)
-        line_statuses2 = dict(self.cobertura2.line_statuses(filename))
+        if self.cobertura2.has_file(filename) and self.cobertura2.filesystem.has_file(
+            filename
+        ):
+            lines2 = self.cobertura2.source_lines(filename)
+            line_statuses2 = dict(self.cobertura2.line_statuses(filename))
+            nonexistent = False
+        else:
+            lines2 = []
+            line_statuses2 = {}
+
+        if nonexistent:
+            # try to get source lines anyway, to get the exception traceback
+            self.cobertura2.source_lines(filename)
```

Comment:
Why is getting the exception useful here? Can you clarify?

Reply:
I guess i wanted the exception in the case that the file doesn't exist in the filesystem of either cobertura1 or cobertura2. i guess the following code would be equivalent (but i'm not sure if it's more clear)
```suggestion
            self.cobertura2._raise_MissingFileSystem(filename)
```

Reply:
Maybe i got tripped up by the fact that `MissingFileSystem` is a sub-class inside the `Cobertura` class

Reply:
It's fine. I looked at the code a little closer and what you did makes sense. Thanks!